testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeColumnMappingMode.java

```diff
@@ -829,6 +827,59 @@ public void testUnsupportedOperationsColumnMappingMode(String mode)
         }
     }
 
+    @Test(groups = {DELTA_LAKE_DATABRICKS, DELTA_LAKE_OSS, DELTA_LAKE_EXCLUDE_73, DELTA_LAKE_EXCLUDE_91, PROFILE_SPECIFIC_TESTS}, dataProvider = "columnMappingWithTrueAndFalseDataProvider")
+    @Flaky(issue = DATABRICKS_COMMUNICATION_FAILURE_ISSUE, match = DATABRICKS_COMMUNICATION_FAILURE_MATCH)
+    public void testOptimizeProcedureColumnMappingMode(String mode, boolean partitioned)
+    {
+        String tableName = "test_dl_optimize_column_mapping_mode_" + randomNameSuffix();
+
+        onDelta().executeQuery("" +
+                "CREATE TABLE default." + tableName +
+                "(a_number INT, a_struct STRUCT<x: INT>, a_string STRING) " +
+                "USING delta " +
+                (partitioned ? "PARTITIONED BY (a_string)" : "") +
+                "LOCATION 's3://" + bucketName + "/databricks-compatibility-test-" + tableName + "'" +
+                "TBLPROPERTIES ('delta.columnMapping.mode'='" + mode + "')");
+
+        try {
+            onTrino().executeQuery("INSERT INTO delta.default." + tableName + " VALUES (1, row(11), 'a')");
+            onTrino().executeQuery("INSERT INTO delta.default." + tableName + " VALUES (2, row(22), 'b')");
+            onTrino().executeQuery("INSERT INTO delta.default." + tableName + " VALUES (3, row(33), 'c')");
+
+            Double stringColumnSize = partitioned ? null : 3.0;
+            List<Row> expectedStats = ImmutableList.<Row>builder()
+                    .add(row("a_number", null, 3.0, 0.0, null, "1", "3"))
+                    .add(row("a_struct", null, null, null, null, null, null))
+                    .add(row("a_string", stringColumnSize, 3.0, 0.0, null, null, null))
+                    .add(row(null, null, null, null, 3.0, null, null))
+                    .build();
+            assertThat(onTrino().executeQuery("SHOW STATS FOR delta.default." + tableName))
+                    .containsOnly(expectedStats);
+
+            // Execute OPTIMIZE procedure and verify that the statistics is preserved and the table is still writable and readable
+            onTrino().executeQuery("ALTER TABLE delta.default." + tableName + " EXECUTE OPTIMIZE");
+
+            assertThat(onTrino().executeQuery("SHOW STATS FOR delta.default." + tableName))
+                    .containsOnly(expectedStats);
+
+            onTrino().executeQuery("INSERT INTO delta.default." + tableName + " VALUES (4, row(44), 'd')");
+            onDelta().executeQuery("INSERT INTO default." + tableName + " VALUES (5, named_struct('x',55), 'e')");
```

Comment:
We could change one of the modification to `UPDATE` to check also other type of modification.

Reply:
I will handle in follow-up.